ddtrace/tracer/span.go

```diff
@@ -319,9 +319,58 @@ func (s *span) finish(finishTime int64) {
 		// not sampled by local sampler
 		return
 	}
+	if t, ok := internal.GetGlobalTracer().(*tracer); ok {
+		// we have an active tracer
+		feats := t.features.Load()
+		if feats.Stats && shouldComputeStats(s) {
+			// the agent supports computed stats
+			var statusCode uint32
+			if sc, ok := s.Meta["http.status_code"]; ok && sc != "" {
+				if c, err := strconv.Atoi(sc); err == nil {
+					statusCode = uint32(c)
+				}
+			}
+			cfg := t.config
+			t.stats.In <- &spanSummary{
+				Start:      s.Start,
+				Duration:   s.Duration,
+				Name:       s.Name,
+				Resource:   s.Resource,
+				Service:    s.Service,
+				Type:       s.Type,
+				Hostname:   s.Meta[keyHostname],
+				Synthetics: strings.HasPrefix(s.Meta[keyOrigin], "synthetics"),
+				Env:        s.Meta[ext.Environment],
+				StatusCode: statusCode,
+				Version:    cfg.version,
+				TopLevel:   s.Metrics[keyTopLevel] == 1,
+				Weight:     s.context.trace.weight(),
```

Comment:
you don't any weight here as you should be processing all the spans in the tracer.
This is one big win of the client stats, no more stats upscaling

Reply:
So just remove and +1 everything?

Reply:
yes